tests/unit/query_plan.cpp

```diff
@@ -662,6 +662,30 @@ TYPED_TEST(TestPlanner, MatchOptionalMatchWhereReturn) {
   DeleteListContent(&optional);
 }
 
+TYPED_TEST(TestPlanner, MatchOptionalMatchNodePropertyWithIndex) {
+  // Test MATCH (n) OPTIONAL MATCH (m) WHERE n.prop = m.prop RETURN n
```

Comment:
Both `n` and `m` should have label in the query, aren't they? You implemented the right thing, but this comment is a bit incomplete.

Reply:
yes that's true, my mistake